<a href="https://colab.research.google.com/github/mdilapi2020/IST210/blob/master/FinalGroupAssignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Auto Shop Database

## Introduction
Our group decided to create an auto part company. This store allows for various products to be sold to the customers. In regards to this project, we made products such as windshield wiper fluid, batteries and such in order to keep it as simplistic as possible. Each one of these products came with a price and where they would be in regards to the store's location (Aisles) labeled accordingly. Another table we created includes employees, each with their respected employee identification number and employee email address so each employee can go into work and clock in or out respectively. With this in mind, we also created a customer table so regulars can create an account for various deals in the future.

In [0]:
%load_ext sql

In [0]:
%sql sqlite:///AutoShop.db

'Connected: @AutoShop.db'

In [0]:
%%sql
CREATE TABLE Aisles(AislesNum text, ItemType text, Primary Key (AislesNum,ItemType));

 * sqlite:///AutoShop.db
Done.


[]

In [0]:
%%sql
INSERT INTO Aisles VALUES ('Aisles 1', 'Oil Filters');
INSERT INTO Aisles VALUES ('Aisles 2', 'Alternators');
INSERT INTO Aisles VALUES ('Aisles 3', 'Wiper Blades');
INSERT INTO Aisles VALUES ('Aisles 4', 'Batteries');
INSERT INTO Aisles VALUES ('Aisles 5', 'Wiper Fluid');

 * sqlite:///AutoShop.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

In [0]:
%%sql
CREATE TABLE Items (UPC text, AislesNum text, Make text, ItemType text ,Price float, Primary Key (UPC, AislesNum, ItemType));

 * sqlite:///AutoShop.db
Done.


[]

In [0]:

%%sql
INSERT INTO Items VALUES ('123456789', '1', 'Presco', 'Oil Filters' ,6.99);
INSERT INTO Items VALUES ('123456781', '2', 'Wenso','Alternators',193.99);
INSERT INTO Items VALUES ('123456782', '3', 'Viper','Wiper Blades', 27.99);
INSERT INTO Items VALUES ('123456783', '4', 'Curacell','Batteries' ,175.99);
INSERT INTO Items VALUES ('123456784', '5', 'LiquidPro','Wiper Fluid' ,2.99);


 * sqlite:///AutoShop.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

In [0]:
%%sql
CREATE TABLE Employee (EmployeeID text Primary Key, FName text, LName text, Email text, PhoneNum text);

 * sqlite:///AutoShop.db
Done.


[]

In [0]:
%%sql
INSERT INTO Employee VALUES ('123', 'John', 'Doe', 'Jdoe123@gmail.com', '914-555-1234');
INSERT INTO Employee VALUES ('124', 'Mike', 'Ike', 'MikeI124@gmail.com', '914-552-1239');
INSERT INTO Employee VALUES ('125', 'Dan', 'Smith', 'DSmith44@aol.com', '914-987-9090');
INSERT INTO Employee VALUES ('162', 'Ryan', 'Burke','Rburke912@verizon.net', '914-989-2112');
INSERT INTO Employee VALUES ('456', 'Colleen', 'Thatcher', 'ColleenT123@gmail.com', '917-181-1921');

 * sqlite:///AutoShop.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

In [0]:
%%sql
Create TABLE Cart(TransactionID text, ItemID int, UPC text, CustomerID text, EmployeeID text, Primary Key (TransactionID, ItemID, UPC,CustomerID,EmployeeID));

 * sqlite:///AutoShop.db
Done.


[]

In [0]:
%%sql
INSERT INTO Cart VALUES ('1', 1, '123456789', '422', '123');
INSERT INTO Cart VALUES ('1', 2, '123456781', '422', '123');
INSERT INTO Cart VALUES ('2', 1, '123456782', '321', '125');
INSERT INTO Cart VALUES ('2', 2, '123456783', '321', '125');
INSERT INTO Cart VALUES ('2', 3, '123456784', '321', '125');


 * sqlite:///AutoShop.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

In [0]:
%%sql
CREATE TABLE Customer(CustomerID text Primary Key, CustomerFName text, CustomerLName text, PhoneNum text);

 * sqlite:///AutoShop.db
Done.


[]

In [0]:
%%sql
INSERT INTO Customer VALUES('321', 'Dan', 'Grossman', '914-420-9876');
INSERT INTO Customer VALUES ('422', 'Mike', 'Leva', '914-321-4312');
INSERT INTO Customer VALUES ('213', 'Robert', 'Dinota', '914-456-7642');
INSERT INTO Customer VALUES ('452', 'Ryan', 'Hayes', '914-391-1818');
INSERT INTO Customer VALUES ('765', 'Jess', 'Woods', '914-391-3242');



 * sqlite:///AutoShop.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

We used our simple ER Diagram in order to ensure that when creating the tables and implementing data it would be less of a burden on the team. Our process for creating data was one person would look around at real auto parts stores, like Autozone or Pep Boys, and then pool real prices for the parts we wanted to include. Next, another person created fake names, phone numbers, and emails for the employees and customers. From there the person in charge of creating the tables would use the command INSERT INTO *Table* VALUES in order to upload these fictitious values.

In [0]:
%%sql
Select I.ItemType, I.Price, A.AislesNum from Items as I, Aisles as A where I.ItemType = A.ItemType and I.ItemType like 'Oi%'

 * sqlite:///AutoShop.db
Done.


[('Oil Filters', 6.99, 'Aisles 1')]

This Query is helpful to the customer. The reason is because it allows the customer to understand where we keep our oil filters and how much they are going to need to spend. In addition, this query is useful for the employee as it can save him or her from fumbling around trying to find the right aisles if they do not have them memorized.

In [0]:
%%sql
Select Count(ItemType) from Aisles where ItemType ='Oil Filters' and ItemType>1

 * sqlite:///AutoShop.db
Done.


[(1,)]

This specific query is helpful to both the employee and the customer. The benefit, or in some cases detriment, of this query is that if a customer is in a rush and needs to find this specific item they know if we have any in stock or not from there they can decide if they need to find another autoparts store. The employee also knows that if the item is no longer in stock then he needs to alert management to reorder more. 

In [0]:
%%sql
Select Count(C.ItemID), SUM(I.Price) from Cart as C, Items as I where C.UPC= I.UPC

 * sqlite:///AutoShop.db
Done.


[(5, 407.95000000000005)]

This query's intended purpose is to create a total for the amount of items sold, in this case 5, and the total amount of money made for the day (407.95). This specific query is helpful to the manager from our scenario who is in charge of keeping a tab on totals to make sure nothing was stolen and the store was not losing money.

> Indented block



![alt text](https://raw.githubusercontent.com/mdilapi2020/IST210/master/pictures/AutoShop.jpg)

Our final ER diagram is still as simple as our previous iterations with some minor, yet necessary fixes in order to ensure a working database. For starters, we liked the idea of having a cart table because it allowed us to count the amount of items per transaction, and to keep tabs as to how many transactions per day. The cart table played nicely with our items table as we could join the two to add the prices for the total amount of transactions made in a day. The fix we made kept our table simple, yet an intuitive design.